### DISCUSSION TITLE
Repository Access Issue

### DISCUSSION THREAD

**Discussion by Ujoa** [community/community#138929] [GitHub Education]
### Hi everyone,

As of the last 15 minutes, students are reporting losing access to their repositories. 

After investigating their repos, they are indeed removed as collaborators. What is going on??
![image](https://github.com/user-attachments/assets/54f6140f-ccbc-4606-94c0-01255f718997)

**Comment by viz26** [comment]
same issue i am getting

**Comment by techibis** [comment]
I am also having this issue and have been adding students as collaborators as a workaround for now.

**Comment by jdelfino** [comment]
I also just encountered this. 2 students out of ~45 reported this in the last hour. I added them back as collaborators. I checked a few other students in the same class, and they still had access, so it isn't a universal problem.

**Reply by Ujoa** [reply]
Accepted students are fine, new **accepting** students are not. Already had 20 email me in the last hr trying to accept

**Reply by jdelfino** [reply]
Good observation, this does fit with what I saw - the students that didn't have access had just accepted an assignment.

**Comment by JoshIsomLehigh** [comment]
I'm having this issue too.  Please submit a trouble ticket at https://support.github.com/ if you also are having this problem.

**Comment by MortenSchou** [comment]
I'm having the same issue just now. 
I notice that students, who now accept assignments, gets their role changed from 'Outside collaborator' to 'Member' in the organization that has all the repositories. And the (default?) setting in the organization for Member privileges -> Base permissions is 'no permission'. I guess this is why they don't get access to their repository. Any ideas how to fix this, without giving the students write access to all other students repositories?
A temporary fix is to add them back as collaborators on their respective repositories. But doing that for 150+ students for multiple different assignments is not viable for me...

**Reply by Wakeful-Cloud** [reply]
I through together a Deno script and prebuilt binaries [here](https://github.com/CSCI-442-Mines/github-classroom-fixes).

**Reply by MortenSchou** [reply]
> I through together a quick workaround [here](https://github.com/CSCI-442-Mines/github-classroom-fixes).

Thanks. 
I ended up making a similar script in bash using [Github Classroom CLI](https://docs.github.com/en/education/manage-coursework-with-github-classroom/teach-with-github-classroom/using-github-classroom-with-github-cli) to [add collaborators to repositories](https://docs.github.com/en/rest/collaborators/collaborators?apiVersion=2022-11-28#add-a-repository-collaborator):
`gh api -XPUT /repos/{your-org}/{your-repo}/collaborators/{your-user}`
This works for me, since Github classroom manages to create the repository and add to the list, but fails to add the student as a collaborator.

I pull the list of repositories using github classroom CLI and generate a bash script with `gh api` commands.
For anyone interested, here is my quick'n'dirty code:
File `add-collaborator.py` contains:
```python
import sys
for line in sys.stdin:
    parts = line.split('\t')
    user = parts[-2].strip()
    url = parts[-1].strip()
    urlparts = url.split('/')
    org = urlparts[-2]
    repo = urlparts[-1]
    print(f'gh api -XPUT /repos/{org}/{repo}/collaborators/{user}')
```

File `add-collaborator.sh` contains:
```bash
#!/bin/bash
assignmentID="$1"
if [ -z "${assignmentID}" ]; then
  echo "Usage: ./add-collaborator.sh <assignemntID> <filename>"
  gh classroom assignments
  exit 1
fi
filename="$2"
if [ -z "${filename}" ]; then
  filename="api-commands.sh"
fi
echo "#!/bin/bash" > ${filename}
chmod +x ${filename}
i=1
while : ; do
  command="gh classroom accepted-assignments -a ${assignmentID} --per-page 30 --page $i"
  echo "Running: ${command}"
  accepted_repos=$(eval "${command}" | tail -n+5)
  [[ -z "${accepted_repos}" ]] && break
  python3 ./add-collaborator.py <<< "${accepted_repos}" >> ${filename}
  ((i++))
done
echo "Check commands in ${filename}"
echo "Then run using ./${filename}"
```

Run this using `./add-collaborator.sh <assignemntID>` then run the gh api commands in `./api-commands.sh`

**Reply by chrissphinx** [reply]
@MortenSchou does this add them as "outside collaborators" though, or direct collaborators? It seems there is no way in the GitHub UI to add students as outside collaborators, which causes them to be added to the organization member list whereas before that wasn't necessary

**Reply by MortenSchou** [reply]
@chrissphinx no, this just adds them as direct collaborators on their reposotory. It does not change their role (member / outside collaborator) in the organisation. But with member base permission set to 'no permission', I don't think there is much practical difference (at least I'm not sure what).
By the way, when I had my class (14 hours ago now), the students had no problem accessing newly accepted assignments, so I didn't really get to use my scripts 😆

**Comment by chrissphinx** [comment]
Any updates @RyanHecht ? Don't want to send out assignment links if nobody will be able to access their repo

**Comment by svlasnik-uno** [comment]
Same issue. Any solution yet?

**Reply by Ujoa** [reply]
Crickets from github 😢

**Comment by Alirico** [comment]
Hello, I am a student, and I have this issue, do you have any suggestions for students like me?

**Comment by dgmouris** [comment]
@RyanHecht I was wondering if you folks have an update on this.

I've been adding students for the last few days on assignments and would love not to have to manually add them, or have them kicked from their repository at what seems like a random time.

Again, I love the product and appreciate your folks work. Looking forward to your folks solution ❤️

**Comment by JoshIsomLehigh** [comment]
I just received the following response from GitHub for my support ticket on this issue:

"Thank you for reaching out to GitHub. Our sincere apologies for the difficulty you are encountering. I have received a new update that a recent change has been made in our system, which should likely resolve the reported behavior.
 
When you get a chance, are you able to test this out on your end today, to confirm if it is still occurring? Please note, the resolution will not apply retroactively, so you will want to have the students re-accept the link.
 
You can also choose to issue new invitations to students as well.
 
If still occurring, to troubleshoot, when you get a chance, can you please direct the students to clear their browser caches to see if this resolves the behavior?
 
Does the same thing occur if they try in an incognito browser?"

**Reply by xwu8** [reply]
Apparently, this problem is still there. And it happens to only some students. This is a serious bug in GitHub Classroom and makes it unusable. Don't know why. Do you have any updates on the fix?
